connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -2382,6 +2382,7 @@ DistributedHerderRequest addRequest(Callable<Void> action, Callback<Void> callba
     }
 
     DistributedHerderRequest addRequest(long delayMs, Callable<Void> action, Callback<Void> callback) {
+        callback.recordStage(tickThreadStage);
```

Comment:
Although this prevents stages recorded by other threads from being overwritten, it does introduce a new risk that a tick thread stage gets recorded after this line completes but before the request can be added to the queue. I think this is still an improvement, and probably the best we can do without synchronization.